enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,288 @@
+---
+title: aws-load-balancer-type
+authors:
+  - "@miheer"
+reviewers:
+  - "@wking"  
+  - "@Miciah"
+  - "@frobware"
+  - "@sdodson"
+approvers:
+  - "@wking"  
+  - "@sdodson"
+  - "@frobware"
+api-approvers:
+  - "@deads"
+  - "@Miciah"
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-942"
+creation-date: 2022-06-09
+last-updated: 2019-06-09   
+status: yet to be implemented
+superseded-by:
+  - ""
+---
+
+# Allow Users to specify Load Balancer Type during installation for AWS
+
+## Release Signoff Checklist
+
+- [ x ] Enhancement is `implementable`
+- [ x ] Design details are appropriately documented from clear requirements
+- [ x ] Test plan is defined
+- [ x ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The users would be able to specify the load balancer type either to elb or nlb during installation. 
+This will be used to deploy default ingress controller with user-specified load balancer service for OpenShift clusters
+running on AWS.
+Currently by default CLB/ELB is set by default by the cluster-ingress-operator.
+
+
+## Motivation
+
+Currently the default load balancer type is set to CLB/ELB. One can set default load balancer during installation using [this](https://docs.openshift.com/container-platform/4.10/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws-network-load-balancer.html#nw-aws-nlb-new-cluster_configuring-ingress-cluster-traffic-aws-network-load-balancer).
+However when the default IngressController CR is deleted, default ingress controller is recreated but as an CLB/ELB.
```

Comment:
I'm unclear on the motivation here.  The cluster has a default.  Great.  Users can override the default.  Great.  If users delete the resource that holds their override, the cluster reverts to the default.  Makes sense to me.  Why can't a user that says "hey, I actually wanted that NLB; I regret deleting the IngressController" adjust the replacement IngressController to ask for an NLB?

This enhancement is currently suggesting a harder-to-accidentally-delete location to store an install-time choice, and it seems easier for both the administrator and component maintainers to have fewer moving pieces by having the user say "this IngressController isn't configured how I'd like, so I'll adjust its `spec`".

Reply:
@wking yes but we need the operator to create the ingress controller with the LB type which user/admin desired for default ingress controller. Please check https://issues.redhat.com/browse/NE-799

comment https://issues.redhat.com/browse/NE-799?focusedCommentId=20329711&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-20329711